### PR TITLE
SEARCH-1382 | handle empty shopping cart

### PR DESCRIPTION
**Pull request by bradgarropy**
## Description

Return `null` if the `ShoppingCart` context does not yet exist.

## Related Issue
 [SEARCH-1382](https://jira.corp.magento.com/browse/SEARCH-1382)

## Motivation and Context

A customer may not have created a `ShoppingCart` yet, so the context creator should handle the case where it doesn't exist.

## How Has This Been Tested?

Tested locally with `jest`, and with the `example` directory while monitoring Snowplow events in Kibana.

## Types of changes

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

-   [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [ ] My change requires a change to the documentation.
-   [ ] I have updated the documentation accordingly.
-   [x] I have read the **CONTRIBUTING** document.
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.

### CODE REVIEW

**Comment by bradgarropy** [comment]
The tests are expected to fail until [this SDK PR](https://github.com/adobe/magento-storefront-events-sdk/pull/43) gets merged.

File: tests/handlers/checkout/instantPurchase.test.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    mockShoppingCartCtx,
+} from "../../utils/mocks";
+
+test("sends snowplow event", () => {
+    instantPurchaseHandler(mockEvent);
+
+    expect(trackStructEvent).toHaveBeenCalledTimes(1);
+
+    expect(trackStructEvent).toHaveBeenCalledWith({
+        category: "checkout",
```
Comment (abrhim) [inline tests/handlers/checkout/instantPurchase.test.ts:19]: An this be extracted to an imported variable?

**Review by abrhim — APPROVED** [review]

File: src/events.ts [hunk 2]
```diff
@@ -45,6 +49,8 @@ const subscribeToEvents = (): void => {
 
 const unsubscribeFromEvents = (): void => {
     mse.unsubscribe.addToCart(addToCartHandler);
+    mse.subscribe.instantPurchase(instantPurchaseHandler);
```
Comment (arjunathreya141291) [inline src/events.ts:52]: Should this be unsubscribe?

Comment (bradgarropy) [reply src/events.ts]: Woops!

**Pull request merged**